lighthouse-core/gather/connections/extension.js

```diff
@@ -99,25 +99,32 @@ class ExtensionConnection extends Connection {
 
   /**
    * @override
-   * @param {!string} method
+   * @param {!string} command
    * @param {!Object} params
    * @return {!Promise}
    */
   sendCommand(command, params) {
     return new Promise((resolve, reject) => {
       log.formatProtocol('method => browser', {method: command, params: params}, 'verbose');
       if (!this._tabId) {
-        log.error('No tabId set for sendCommand');
+        log.error('ExtensionConnection', 'No tabId set for sendCommand');
       }
+
       chrome.debugger.sendCommand({tabId: this._tabId}, command, params, result => {
         if (chrome.runtime.lastError) {
-          log.formatProtocol('method <= browser ERR', {method: command, params: result}, 'error');
-          return reject(chrome.runtime.lastError);
-        }
-
-        if (result.wasThrown) {
-          log.formatProtocol('method <= browser ERR', {method: command, params: result}, 'error');
-          return reject(result.exceptionDetails);
+          const message = chrome.runtime.lastError.message;
+          let error;
+          try {
+            error = JSON.parse(message);
+          } catch (e) {}
+          error = error || {message: 'Unknown debugger protocol error.'};
+
+          const callback = {
+            resolve,
```

Comment:
incidentally, this is one more reason we need to always avoid `Promise.all`ing any debugger protocol code since any refactor like this might introduce ticks between steps, even in driver functions that only run a single command :)